File: Chapter 03/Python/section_3_3_5_dip/after/winger.py
from .player import Player

class Winger(Player):
    """
    DIP PROOF: We can add this new player type and inject it into 
    the Coach without modifying a single line of coach.py!
    """
    def perform_action(self):
        print("  [Action] Winger is running down the sideline.")

File: Chapter 03/C#/Section_3_3_5_DIP/After/Winger.cs
using System;

namespace Chapter03.DIP.After;

/// <summary>
/// DIP SOLUTION: The "Plug-and-Play" Proof.
/// 
/// ARCHITECTURE NOTE: Because the Coach depends on an abstraction, we were 
/// able to create this brand new Winger class and instantly inject it into 
/// the Coach's game plan without altering a single line of the Coach class! 
/// 
/// This is the ultimate goal of software architecture: adding new features 
/// by adding new code, rather than rewriting existing code.
/// </summary>
public class Winger : IPlayer
{
    public void PerformAction() => Console.WriteLine("  [Action] Winger is running down the sideline.");
}